model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -173,10 +173,19 @@ object AddressResponseAddress {
 
     val poBoxNumber = if (paf.poBoxNumber.isEmpty) "" else s"PO BOX ${paf.poBoxNumber}"
 
+    val trimmedBuildingNumber = paf.buildingNumber.trim
+    val trimmedDependentThoroughfare = paf.dependentThoroughfare.trim
+    val trimmedThoroughfare = paf.thoroughfare.trim
+
+    val buildingNumberWithStreetName = if (trimmedDependentThoroughfare.isEmpty) {
```

Comment:
You could reduce the amount of code here with using an expression in your interpolated strings.

eg:
```
val buildingNumberWithStreetName = s"$trimmedBuildingNumber  ${if(!trimmedDependentThoroughfare.isEmpty) {s"$trimmedDependentThoroughfare, " }$trimmedThoroughfare"
```